pkg/oauth/middleware.go

```diff
@@ -0,0 +1,47 @@
+package oauth
+
+import (
+	"net/http"
+
+	"github.com/genmcp/gen-mcp/pkg/mcpfile"
+)
+
+// Middleware returns a middleware function that adds OAuth 2.0 Protected Resource Metadata endpoint support
+func Middleware(config *mcpfile.MCPServer) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		httpConfig := config.Runtime.StreamableHTTPConfig
+
+		// Only create OAuth handler if configured
+		if httpConfig.Auth == nil {
+			return next // No OAuth config, just pass through
+		}
+
+		// Convert mcpfile.AuthConfig to oauth.MetadataConfig
+		metadataConfig := MetadataConfig{
+			ResourceName:           config.Name,
+			AuthorizationServers:   httpConfig.Auth.AuthorizationServers,
+			ScopesSupported:        httpConfig.Auth.ScopesSupported,
+			BearerMethodsSupported: httpConfig.Auth.BearerMethodsSupported,
+			JWKSURI:                httpConfig.Auth.JWKSURI,
+			ResourceDocumentation:  httpConfig.Auth.ResourceDocumentation,
+		}
+		oauthHandler := NewProtectedResourceMetadataHandler(metadataConfig)
```

Comment:
Does it makes sense to do this before line 11 (i.e. the `return func...`) so that we don't have to create this on every request the middleware handles?